### ISSUE TITLE
amplify 7.x.x fails ci/cd build for one branch / fine on other

### ISSUE THREAD

**Issue by ingwinlu** [aws-amplify/amplify-cli#8873]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### How did you install the Amplify CLI?

npm

### If applicable, what version of Node.js are you using?

v16.10.0

### Amplify CLI Version

7.x.x

### What operating system are you using?

Ubuntu WSL

### Amplify Categories

auth, storage, function, api, hosting

### Amplify Commands

push

### Describe the bug

Today I was pinged by a former customer that their amplify builds (which i setup and contributed to) where suddenly failing the ci/cd pipeline, sometimes.

The sytstem builds the 'dev' backend on pushes to master, 'prod' on pushes to the 'prod' branch. prod always was fine, on pushes to master we saw the following issue
![amplify_original_error](https://user-images.githubusercontent.com/4435962/141833246-d0a498e8-ea96-4575-af08-60f539853fda.png)
:
```
2021-11-15T15:25:20.701Z [INFO]: UPDATE_COMPLETE amplify-controlapp-dev-95224 AWS::CloudFormation::Stack Mon Nov 15 2021 15:25:20 GMT+0000 (Coordinated Universal Time)
2021-11-15T15:25:30.450Z [WARNING]: ✖ An error occurred when pushing the resources to the cloud
2021-11-15T15:25:30.453Z [WARNING]: ✖ There was an error initializing your environment.
2021-11-15T15:25:30.461Z [INFO]: [0mTypeError: Cannot read property 'includes' of undefined[0m
                                 [0m    at moveBackendResourcesToCurrentCloudBackend (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/src/extensions/amplify-helpers/update-amplify-meta.ts:67:81)[0m
                                 [0m    at AmplifyToolkit.updateamplifyMetaAfterPush [as _updateamplifyMetaAfterPush] (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/src/extensions/amplify-helpers/update-amplify-meta.ts:206:3)[0m
                                 [0m    at runMicrotasks (<anonymous>)[0m
                                 [0m    at processTicksAndRejections (internal/process/task_queues.js:97:5)[0m
                                 [0m    at Object.run (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/src/push-resources.ts:310:7)[0m
2021-11-15T15:25:30.490Z [ERROR]: !!! Build failed
2021-11-15T15:25:30.490Z [ERROR]: !!! Non-Zero Exit Code detected
2021-11-15T15:25:30.490Z [INFO]: # Starting environment caching...
2021-11-15T15:25:30.491Z [INFO]: # Uploading environment cache artifact...
2021-11-15T15:25:30.550Z [INFO]: # Environment caching completed

```
The backend was at this point successfully pushed (all stack successfully deployed) but the build failed and no frontend changes were rolled out.
The amplify push is done via the 'default' ` amplifyPush --simple`.

master and prod at this point are functionally identical (master was merged into prod via git merge --no-ff master). Dev might have been pushed to from a developers workstation via amplify push with a cli (version 7.x.x).

Pinning the build system to use amplify 6.4.0 resolves the build issues and makes the ci behaviour consistent again.

It should also be noted that the amplify project also utilizes custom cloudformation categories / resources (setting up iot rules). From the error it is difficult to say which part exactly did not have content where it was expected.

### Expected behavior

Builds have same build status for same code on both branches when their code is identical.

### Reproduction steps

not sure what first trigger this. I assume amplify cli v7 could be the issue as according to skimming the docs their might be some migration necessary.

### GraphQL schema(s)

<details>

```graphql
# Put schemas below this line


```

</details>

### Log output

<details>

```
# Put your logs below this line


```

</details>

**Comment by kaustavghosh06** [comment]
@ingwinlu Thank you for sending over your amplify folder to us. We took a look at the files and root-caused it to the "iotrule" custom category that you're having in the project which is missing a "service" key in all it's corresponding resources like the other Amplify resources. We're working on a fix right now but in the meanwhile if you add add a "service" to your custom category resources which you have in your project like other Amplify resources, you should be unblocked. We'll post an update once we release the fix for it.
